src/rest_parser.cpp

```diff
@@ -580,8 +580,25 @@ Status KFSRestParser::parseData(rapidjson::Value& node, ::inference::ModelInferR
         HANDLE_VALUE(mutable_uint_contents, GetUint)
     } else if (input->datatype() == "FP64") {
         HANDLE_VALUE(mutable_fp64_contents, GetFloat)
+    } else if (input->datatype() == "BOOL") {
+        for (auto& value : node.GetArray()) {
+            if (value.IsArray()) {
+                for (auto& v : node.GetArray()) {
+                    auto status = parseData(v, input);
+                    if (!status.ok()) {
+                        return status;
+                    }
+                }
+                return StatusCode::OK;
+            }
+            if (!value.IsBool()) {
+                return StatusCode::REST_COULD_NOT_PARSE_INPUT;
+            }
+            input->mutable_contents()->mutable_bool_contents()->Add(value.GetBool());
```

Comment:
Can't we use macro here?

Reply:
Oh I see

Reply:
I implemented that this way because for BOOL there is check if value.isBool() and for others datatypes if value.isNumber(). Of course it could be done by adding another parameter to macro but im not sure if that would be better?